test/method.jl

```diff
@@ -0,0 +1,184 @@
+        @test_signature f6(x::S, y::T) where {T,S} = 2x
+        @test_signature f7(x::S, y::T) where T where S = 2x
+    end
+
+    @testset "Whereparams with constraints" begin
+        @test_signature f8(x::S) where S<:Integer = 2x
+        @test_signature f9(x::S, y::S) where S<:Integer = 2x
+        @test_signature f10(x::S, y::T) where {S<:Integer, T<:Real} = 2x
+        @test_signature f11(x::S, y::Int64) where S<:Integer = 2x
+
+        @test_signature f12(x::S) where {S>:Integer} = 2x
+        @test_signature f13(x::S) where Integer<:S<:Number = 2x
+    end
+
+    @testset "Arg types with type-parameters" begin
+        @test_signature f14(x::Array{Int64, 1}) = 2x
+        @test_signature f15(x::Array{T, 1}) where T = 2x
+
+        @test_signature f16(x::Array{T, 1} where T<:Real) = 2x
+
+        # This is the same method as f16 (other than name), and one displaces the other
+        # but they have different method objects. And different (but equivelent) ASTd
+        # this generates something that should be the same as what `signature(f16)` does
+        # but with a gensym'd variable name
+        f16_alt(x::Array{<:Real, 1}) = 2x
+        f16_alt_sig = signature(only_method(f16_alt))
+        @test f16_alt_sig[:name] == :f16_alt
+        @test occursin(  # Hack to deal with gensymed name. Make it a string and use regex
+            r"^\QExpr[:(x::(Array{\E(.*?)\Q, 1} where \E\1\Q <: Real))]\E$",
+            string(f16_alt_sig[:args])
+        )
+    end
+
+    @testset "anon functions" begin
+        @test_signature (x) -> x  # no args
+        @test_signature (x) -> 2x
+
+        @test_signature ((::T) where T) -> 0   # Anonymous parameter
+
+
+    end
+
+    @testset "vararg" begin
+        @test_signature f17(xs::Vararg{Any, N} where N) = 2
+
+        # `f17_alt(xs...) = 2` lowers to the same method as `f18`
+        # but has a different AST according to `splitdef` so we can't us @test_signature
+        f17_alt(xs...) = 2
+        test_matches(
+            signature(only_method(f17_alt)),
+            Dict(
+                :name => :f17_alt,
+                :args => [:(xs::(Vararg{Any, N} where N))]
+            )
+        )
+
+        @test_signature f18(xs::Vararg{Int64, N} where N) = 2
+        @test_signature f19(x, xs::Vararg{Any, N} where N) = 2x
+    end
+
+    @testset "kwargs" begin  # We do not support them right now
+        #Following is broken:
+        #@test_signature kwargs17(x; y=3x) = 2x
+        kwargs17(x; y=3x) = 2x
+        # at least be sure we get the rest right:
+        test_matches(
+            signature(only_method(kwargs17)),
+            Dict(
+                :name => :kwargs17,
+                :args => [:x]
+            )
+        )
+    end
+
+    # Only test on 1.3 because of issues with declaring structs in 1.0-1.2
+    # TODO: https://github.com/invenia/ExprTools.jl/issues/7
+    VERSION >= v"1.3" && @testset "Constructors (basic)" begin
```

Comment:
I have just disabled this test because in 1.0 it is fiddly to get it working right,
and I have openned an issue about it.
Is this ok?

Reply:
Make an issue for this as we should get it working on the LTS version of Julia

Reply:
https://github.com/invenia/ExprTools.jl/issues/7